crates/sui-types/src/crypto.rs

```diff
@@ -277,77 +290,220 @@ impl std::fmt::Debug for Signature {
     }
 }
 
-impl Signature {
-    pub fn new<T>(value: &T, secret: &dyn signature::Signer<Signature>) -> Self
-    where
-        T: Signable<Vec<u8>>,
-    {
-        let mut message = Vec::new();
-        value.write(&mut message);
-        secret.sign(&message)
+//
+// Ed25519 Sui Signature port
+//
+#[serde_as]
+#[derive(Clone, Debug, Serialize, Deserialize, JsonSchema)]
+pub struct Ed25519SuiSignature(
+    #[schemars(with = "Base64")]
+    #[serde_as(as = "Readable<Base64, Bytes>")]
+    [u8; Ed25519PublicKey::LENGTH + Ed25519Signature::LENGTH + FLAG_LENGTH],
+);
+
+impl SuiSignatureInner for Ed25519SuiSignature {
+    type Sig = Ed25519Signature;
+    type PubKey = Ed25519PublicKey;
+    type KeyPair = Ed25519KeyPair;
+    const LENGTH: usize = Ed25519PublicKey::LENGTH + Ed25519Signature::LENGTH + FLAG_LENGTH;
+}
+
+impl SuiPublicKey for Ed25519PublicKey {
+    const FLAG: [u8; FLAG_LENGTH] = [0xed];
```

Comment:
As we agreed, we should set the FLAG for ed25519 to = 0x00. We're not going with multicodec standard due to variable length.

Reply:
removed.